octree/include/pcl/octree/octree_iterator.h

```diff
@@ -135,19 +114,18 @@ namespace pcl
          */
         bool operator==(const OctreeIteratorBase& other) const
         {
-          return (( octree_ ==other.octree_) &&
-                  ( current_state_ == other.current_state_) &&
-                  ( max_octree_depth_ == other.max_octree_depth_) );
+          return (this == &other) ||
+                    ((octree_ == other.octree_) &&
+                     (current_state_ == other.current_state_) &&
```

Comment:
Then here we should dereference, right? Otherwise this is never true.
(Plus protection from dereferencing null pointer).